src/main/kotlin/org/hazelcast/configdsl/HazelcastDsl.kt

```diff
@@ -45,12 +42,12 @@ fun QueryCacheConfig.entryListener(implementation: MapListener, local: Boolean,
     EntryListenerConfig(implementation, local, includeValue)
         .apply { this@entryListener.addEntryListenerConfig(this) }
 
-fun QueryCacheConfig.index(name: String,
-                           ordered: Boolean = false,
-                           init: MapIndexConfig.() -> Unit = {}) =
-    MapIndexConfig(name, ordered)
-        .apply(init)
-        .apply { this@index.addIndexConfig(this) }
+fun QueryCacheConfig.index(type: IndexType = IndexType.SORTED,
```

Comment:
I'm afraid the original code was not correct to start with

Reply:
Default parameter values are a great idea!

Reply:
> I'm afraid the original code was not correct to start with

can you guide me what is the correct thing to do in that case?

Reply:
> can you guide me what is the correct thing to do in that case?

Forget my previous comment. It's fine.

Reply:
okay